### PR TITLE
Add method_ref, uri_ref, headers_ref to request::Builder

### PR DESCRIPTION
**Pull request by kazuk**
implementation for https://github.com/hyperium/http/issues/282

### CODE REVIEW

File: src/request.rs [hunk 1]
```diff
@@ -771,6 +771,34 @@ impl Builder {
...
+    /// # use http::*;
+    /// 
+    /// let mut req = Request::builder();
+    /// assert_eq!(req.get_method(),Some(Method::GET));
+    /// req.method("POST");
+    /// assert_eq!(req.get_method(),Some(Method::POST));
+    /// req.method("DELETE");
+    /// assert_eq!(req.get_method(),Some(Method::DELETE));
+    /// ```
+    pub fn get_method(&self) -> Option<Method>
```
Comment (carllerche) [inline src/request.rs]: I'd probably return a reference here.

File: src/request.rs [hunk 2]
```diff
@@ -860,6 +913,33 @@ impl Builder {
...
+    /// # use http::header::HeaderValue;
+    /// # use http::request::Builder;
+    /// let mut req = Request::builder();
+    /// req.header("Accept", "text/html")
+    ///    .header("X-Custom-Foo", "bar");
+    /// let headers = req.get_headers().unwrap();
+    /// assert_eq!( headers["Accept"], "text/html" );
+    /// assert_eq!( headers["X-Custom-Foo"], "bar" );
+    /// ```
+    pub fn get_headers(&self ) -> Option<HeaderMap<HeaderValue>> {
```
Comment (carllerche) [inline src/request.rs]: Extra space.

File: src/request.rs [hunk 3]
```diff
@@ -771,6 +771,34 @@ impl Builder {
...
+    /// assert_eq!(req.get_method(),Some(Method::POST));
+    /// req.method("DELETE");
+    /// assert_eq!(req.get_method(),Some(Method::DELETE));
+    /// ```
+    pub fn get_method(&self) -> Option<Method>
+    {
+        if self.err.is_some() {
+            return None
+        }
+        match &self.head {
```
Comment (seanmonstar) [inline src/request.rs]: You should be able to remove most of these `&`s:

```rust
match self.head {
    Some(ref head) => Some(&head.method),
    None => None,
}
```

File: src/request.rs [hunk 4]
```diff
@@ -800,6 +828,31 @@ impl Builder {
...
+    /// 
+    /// ```
+    /// # use http::*;
+    /// 
+    /// let mut req = Request::builder();
+    /// assert_eq!(req.get_uri().unwrap().to_string(), "/" );
+    /// req.uri("https://www.rust-lang.org/");
+    /// assert_eq!(req.get_uri().unwrap().to_string(), "https://www.rust-lang.org/" );
+    /// ```
+    pub fn get_uri(&self) -> Option<Uri>
```
Comment (seanmonstar) [inline src/request.rs]: This should return an `Option<&Uri>`. The user can decide if they need to clone it, but if not, no need to be wasteful :)

File: src/request.rs (hunk 2 above)
Comment (seanmonstar) [inline src/request.rs]: Should return `Option<&HeaderMap>`

**Pull request merged**